contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -947,17 +947,41 @@ Datum sysutcdatetime(PG_FUNCTION_ARGS)
 
 Datum getutcdate(PG_FUNCTION_ARGS)
 {
-    PG_RETURN_DATUM(DirectFunctionCall2(timestamp_trunc,CStringGetTextDatum("millisecond"),DirectFunctionCall2(timestamptz_zone,CStringGetTextDatum("UTC"),
-                                                            TimestampTzGetDatum(GetCurrentStatementStartTimestamp()))));
-    
+	Datum utc_time;
+    Timestamp rounded_time;
+
+	/* First get UTC time */
+    utc_time = DirectFunctionCall2(timestamptz_zone,
+									CStringGetTextDatum("UTC"),
+									TimestampTzGetDatum(GetCurrentStatementStartTimestamp()));
+
+	/* Convert Datum to Timestamp for roundoff_datetime */
+	rounded_time = (*common_utility_plugin_ptr->roundoff_datetime)(DatumGetTimestamp(utc_time));
```

Comment:
why is this needed?

Reply:
this function returns datetime, which should be rounded. Else it returns original value -- which does not comply with TSQL standards